src/Config/SettingsTab.ts

```diff
@@ -223,6 +223,19 @@ export class SettingsTab extends PluginSettingTab {
                     });
             });
 
+        new Setting(containerEl)
```

Comment:
I don't think 'Date Settings' is the correct location for this. It isn't changing the behaviour of any dates.

Reply:
> In particular, if a user changes their setting to switch the order of adding new recurrence, what code will be broken.

Do you mean if they switch it on, toggle tasks, switch it off, toggle tasks, etc? I just tested this, it is fine. This is just about in which direction from the original task are we writing the completed tasks.

> There is code all over Tasks that assumes that the new recurrence is the first task returned.
>
> It's clearly documented in this method that that is the behaviour.
>
> I am afraid changing this behaviour is a very high risk change. And it will mean that every place in future where this method is called will have to check the settings to know which is the old task and which is the new.
>
> I think the correct place for the reversing of tasks is the point where the toggled lines are written to the note, so that this PR's change is much more localised.

This is a very good point indeed, I have not considered this. I think I know a place to fit this but then I have doubts on how to write the tests. Let me think about it tomorrow and if I won't find a solution, let's discuss on a pairing session.

> I don't think 'Date Settings' is the correct location for this. It isn't changing the behaviour of any dates.

Agree. I have not found a meaningful place for this to fit, so I'm open for suggestions. If you don't wish to spend time on this typing we can tackle this on the next pairing session too as this is pretty quick.

Reply:
Definitely a good idea to chat when we pair. Much higher bandwidth than my hasty, terse comments so far.  😄